tests/DoctrineObjectTest.php

```diff
@@ -2866,4 +2927,58 @@ public function testNestedHydrationByReference(): void
         $this->assertSame('value', $entity->getToOne(false)->getField(false));
         $this->assertSame('2019-01-24 12:00:00', $entity->getCreatedAt()->format('Y-m-d H:i:s'));
     }
+
+    public function testHandleEnumConversionUsingByValue(): void
+    {
+        if (PHP_VERSION_ID < 80100) {
+            $this->markTestSkipped('PHP 8.1 required for enum compatibility');
```

Comment:
I think PHPUnit has `@requires php >= 8.1` or such?

See https://phpunit.readthedocs.io/en/9.5/incomplete-and-skipped-tests.html#incomplete-and-skipped-tests-requires-tables-api